packages/pwa-buildpack/lib/WebpackTools/configureWebpack.js

```diff
@@ -0,0 +1,342 @@
+                        path.join(moduleDir, 'src', 'RootComponents'),
+                        path.join(moduleDir, 'lib', 'RootComponents')
+                    ],
+                    []
+                ),
+                context
+            }),
+            new webpack.EnvironmentPlugin(projectConfig.env),
+            new ServiceWorkerPlugin({
+                mode,
+                paths,
+                injectManifest: true,
+                injectManifestConfig: {
+                    include: [/\.js$/],
+                    swSrc: './src/sw.js',
+                    swDest: './sw.js'
+                }
+            }),
+            new UpwardIncludePlugin({
+                upwardDirs: hasFlag('upward')
+            }),
+            new WebpackAssetsManifest({
+                output: 'asset-manifest.json',
+                entrypoints: true,
+                publicPath: '/',
+                // Add explicit properties to the asset manifest for
+                // upward.yml to use when evaluating app shell templates.
+                transform(assets) {
+                    // All RootComponents go to prefetch, and all client scripts
+                    // go to load.
+                    assets.bundles = {
+                        load: assets.entrypoints.client.js,
+                        prefetch: []
+                    };
+                    Object.entries(assets).forEach(([name, value]) => {
+                        if (name.match(/^RootCmp.*\.js$/)) {
+                            const filenames = Array.isArray(value)
+                                ? value
+                                : [value];
+                            assets.bundles.prefetch.push(...filenames);
+                        }
+                        const ext = path.extname(name);
+                        const type = ext && ext.replace(/^\./, '');
+                        if (type) {
+                            if (!assets[type]) {
+                                assets[type] = {};
+                            }
+                            assets[type][path.basename(name, ext)] = value;
+                        }
+                    });
+                }
+            })
+        ]
+    };
+    let vendorTest = '[\\/]node_modules[\\/]';
+    if (vendor.length > 0) {
+        vendorTest += `(${vendor.join('|')})[\\\/]`;
+    }
+    config.optimization = {
+        splitChunks: {
+            cacheGroups: {
+                vendor: {
+                    test: new RegExp(vendorTest),
+                    chunks: 'all'
+                }
+            }
+        }
+    };
+    if (debugBundles) {
+        Object.assign(config.optimization, {
+            moduleIds: 'named',
+            nodeEnv: 'development',
+            minimize: false,
+            occurrenceOrder: true,
+            usedExports: true,
+            concatenateModules: true,
+            sideEffects: true
+        });
+    } else if (mode === 'development') {
```

Comment:
[Earlier in this file](https://github.com/magento-research/pwa-studio/pull/1498/files#diff-6cf0920ab2bb3105870a4067a11707f4R94) if the debugBundles variable is true we set `mode` to "development".

This conditional suggests that it is possible to be in development mode without `debugBundles`. Perhaps this and the above block should be in a single conditional `if (debugBundles || mode === 'development') { ...`